### PR TITLE
Avoid CI runs under users' own CircleCI personal account

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Sometimes the tests will run in the CircleCI of the user and not run our tests since they don't have access to our resources. One example is on [this PR](https://github.com/huggingface/transformers/pull/20479#issuecomment-1369690668) where the "real" tests were not run.

**We can make the new job `check_circleci_user` required (too) - once this PR is merged into `main`**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -9,6 +9,19 @@ parameters:
...
+        docker:
+            - image: cimg/python:3.7.12
+        parallelism: 1
+        steps:
+            - run: echo $CIRCLE_PROJECT_USERNAME
+            - run: |
+                if [ "$CIRCLE_PROJECT_USERNAME" = "huggingface" ]; then
+                    exit 0
+                else
+                    exit -1
```
Comment (sgugger) [inline .circleci/config.yml:23]: Maybe echo something here so that when reading the log, we understand what failed?

**Review by sgugger — APPROVED** [review]
Thanks for working on this! Just add one message to the exception and it should be good to go!

File: .circleci/config.yml [hunk 2]
```diff
@@ -9,6 +9,19 @@ parameters:
         default: false
 
 jobs:
+    # Ensure running with CircleCI/huggingface
+    check_circleci_user:
+        docker:
+            - image: cimg/python:3.7.12
+        parallelism: 1
+        steps:
+            - run: echo $CIRCLE_PROJECT_USERNAME
```
Comment (ydshieh) [inline .circleci/config.yml:18]: @sgugger  it's here :-)

File: .circleci/config.yml (hunk 1 above)
Comment (ydshieh) [reply .circleci/config.yml:23]: Oh, I see, more clear message here.

Comment (ydshieh) [reply .circleci/config.yml]: It's now looks like (if error occurs)

```
The CI is running under the huggingface personal account. Please following https://support.circleci.com/hc/en-us/articles/360008097173-Troubleshooting-why-pull-requests-are-not-triggering-jobs-on-my-organization- to fix it.

Exited with code exit status 255
CircleCI received exit code 255
```

**Comment by lishukan** [comment]
@ydshieh  Unfortunately, I encountered the problem you mentioned last night and I don't know how to solve it. The link provided in your MR ( [https://support.circleci.com/hc/en-us/articles/360008097173-Troubleshooting-why-pull-requests-are-not-triggering-jobs-on-my-organization- ](https://support.circleci.com/hc/en-us/articles/360008097173-Troubleshooting-why-pull-requests-are-not-triggering-jobs-on-my-organization-) ) has expired. Can you tell me how to solve this problem.? This is my PR [#25519](https://github.com/huggingface/transformers/pull/25519 )  After I submitted it, the circle ci began to execute automatically, but this issue occurred afterwards.

**Pull request merged**